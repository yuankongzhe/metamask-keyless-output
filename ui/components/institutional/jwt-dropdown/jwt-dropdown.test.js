import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import sinon from 'sinon';
import JwtDropdown from './jwt-dropdown';

describe('JwtDropdown', () => {
  it('should render the Jwt dropdown component', () => {
    const props = {
      jwtList: ['jwy1', 'jwt2'],
      currentJwt: 'someToken',
      onChange: sinon.spy(),
    };

    const { getByTestId, container } = render(<JwtDropdown {...props} />);

    fireEvent.change(getByTestId('jwt-dropdown'), {
      target: { value: 'jwt2' },
    });

    expect(getByTestId('jwt-dropdown')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
