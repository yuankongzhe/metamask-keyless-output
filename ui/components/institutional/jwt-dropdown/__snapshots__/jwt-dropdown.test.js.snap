// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JwtDropdown should render the Jwt dropdown component 1`] = `
<div>
  <div
    class="mm-box"
  >
    <p
      class="mm-box mm-text custody-search-jwt__select-title mm-text--body-md mm-box--padding-1 mm-box--color-text-default"
    >
      [selectJWT]
    </p>
    <div
      class="dropdown custody-search-jwt__select"
    >
      <select
        class="dropdown__select"
        data-testid="jwt-dropdown"
      >
        <option
          value="someToken"
        >
          someToken
        </option>
        <option
          value="jwy1"
        >
          ...jwy1
        </option>
        <option
          value="jwt2"
        >
          ...jwt2
        </option>
      </select>
      <span
        class="mm-box dropdown__icon-caret-down mm-icon mm-icon--size-sm mm-box--display-inline-block mm-box--color-inherit"
        style="mask-image: url('./images/icons/arrow-down.svg');"
      />
    </div>
  </div>
</div>
`;
